### PR TITLE
vscode-extensions.vadimcn.vscode-lldb: 1.9.1 -> 1.9.2

### LINKED ISSUE #160874: codelldb inside of vscode-lldb extension doesn't work
### Describe the bug
I want to use `vscode-extensions.vadimcn.vscode-lldb.adapter` (providing `codelldb`) for it's Debug Adapter support. Particularly it seems like Helix can utilize it with some early debugging support.

However, it doesn't seem to work. After adding that to my nix shell, and entering it, and trying to run it, I get:

```
[cole@porty:~/code/rust-nix-template]$ codelldb
Error: "\"/nix/store/rnqda0vmhg95ziahvl7r6y10x4xk1y6z-vscode-lldb-adapter-1.6.10/lldb/lib/liblldb.so: cannot open shared object file: No such file or directory\""
```

### LINKED ISSUE #234908: vscode-lldb: libcodelldb.so not found
### Describe the bug
see title

### Steps To Reproduce
1. install the extension using nix
2. try to use it
3. fail

### Expected behavior
The extension works fine.

### Notify maintainers
@nigelgbanks 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.3.1-zen1, NixOS, 23.11 (Tapir), 23.11pre488026.f91ee3065de`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos"`
 - channels(pearsche): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by Congee**
The patched adapter now produces a dynamic library instead of a static archive

fixes #234908
fixes #160874

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Congee** [comment]
Result of `nixpkgs-review pr 236915` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
  </ul>
</details>

**Comment by Congee** [comment]
lldb version remains at 14.0.6 as I am unable to build llvm 15.0.7 from nixpkgs against the 15.x source code which seems to be exactly 15.0.0 at https://github.com/vadimcn/llvm-project/tree/codelldb/15.x

The error was
> mismatch in the patch version! we have version 15.0.7 and expected the patch version to be '7'; the source has '0' instead

File: pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix [hunk 1]
```diff
@@ -15,7 +15,7 @@ let
     owner = "vadimcn";
     repo = "vscode-lldb";
     rev = "v${version}";
-    sha256 = "sha256-DqxdZtSW8TZaOFGXOZQ7a4tmgRj6iAWDppCNomdfVxY=";
+    sha256 = "6QmYRlSv8jY3OE3RcYuZt+c3z6GhFc8ESETVfCfF5RI=";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix:18]:
```suggestion
    hash = "sha256-6QmYRlSv8jY3OE3RcYuZt+c3z6GhFc8ESETVfCfF5RI=";
```

File: pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix [hunk 2]
```diff
@@ -25,7 +25,7 @@ let
     pname = "${pname}-adapter";
     inherit version src;
 
-    cargoSha256 = "sha256-+hfNkr9cZbOcWdWKUWUqDj9a0PKjKeApFXYZzS1XokE=";
+    cargoSha256 = "sha256-Qq2igtH1XIB+NAEES6hdNZcMbEmaFN69qIJ+gTYupvQ=";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix:28]:
```suggestion
    cargoHash = "sha256-Qq2igtH1XIB+NAEES6hdNZcMbEmaFN69qIJ+gTYupvQ=";
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix (hunk 2 above)
Comment (figsoda) [reply pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix]: changing from `cargoSha256` to `cargoHash` will change the hash

Comment (SuperSandro2000) [reply pkgs/applications/editors/vscode/extensions/vadimcn.vscode-lldb/default.nix]: really!? sigh

**Pull request merged**